eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs

```diff
@@ -353,19 +357,31 @@ validateCollateralContainsNonADA bal =
 -- > (_,i_f) := txvldt tx
 -- > ◇ ∉ { txrdmrs tx, i_f } ⇒ epochInfoSlotToUTCTime epochInfo systemTime i_f ≠ ◇
 validateOutsideForecast ::
-  HasField "vldt" (Core.TxBody era) ValidityInterval =>
+  ( HasField "vldt" (Core.TxBody era) ValidityInterval,
+    HasField "_protocolVersion" (Core.PParams era) ProtVer
+  ) =>
+  Core.PParams era ->
   EpochInfo (Either a) ->
+  -- | Current slot number
+  SlotNo ->
   SystemStart ->
+  -- | Stability window, in terms of number of slots
+  Word64 ->
   ValidatedTx era ->
   Test (UtxoPredicateFailure era)
-validateOutsideForecast ei sysSt tx =
+validateOutsideForecast pp ei slotNo sysSt sw tx =
```

Comment:
Also: when we disable the check of the validity interval, Plutus still requires we actually provide POSIX times. This currently happens via `collectTwoPhaseScriptInputs` et al, and it uses the `EpochInfo Identity`. And it crashes my custom node when using this patch.

There's a few options that come to mind.

- we either need a Consensus patch that lifts enables arbitrary far ahead translations when appropriate and use that here, when the check has been disabled

- or eg would it be sound to just use a constant, extremely wide/generous "translated" POSIX time?

- or maybe we "translate" the upper bound to the end of the stability window (perhaps unless the upper bound is already within that)?

Reply:
The first option seems the best to me, it is consistent with the current semantics.

Unless I am misunderstanding the second two options, I don't think that they are an option, I do think that they break the Plutus semantics. We have no control what users are doing with the POSIX time inside of a Plutus script, we need to continue to translate the slots to the POSIX time that they are currently being translated to. (eg, imagine a script that succeeds iff the ttl is exactly some particular value).

Reply:
I tried `min slot maxStableSlot` and that got me farther but then failed. I'm guessing the script had some redundancy checks?

Ah ha! I just realized something: such redundancy checks shouldn't don't risk interfering with the Babbage transition because Babbage doesn't change the forecasts (ie slot or epoch length). That's a bit of a relief.